### PR TITLE
[#16] Add pool tests 

### LINKED ISSUE #16: Tests for pool
It's a bit more complex because pool uses `client` NeoFS package so we either need some mocked node or https://github.com/testcontainers/testcontainers-go.

### PR DESCRIPTION
**Pull request by KirillovDenis**
closes #16 

Signed-off-by: Denis Kirillov <denis@nspcc.ru>

### CODE REVIEW

**Review by masterSplinter01 — APPROVED** [review]

**Review by masterSplinter01 — APPROVED** [review]
Re-approve after commiting of missing autogenerated files.

**Review by alexvanin — APPROVED** [review]
I am not a big fan of mockgen, but we can try it out.

**Comment by KirillovDenis** [comment]
> I am not a big fan of mockgen, but we can try it out.

I've tried https://github.com/testcontainers/testcontainers-go. This package would be convenient if we have just one container, but we need to use compose-files.

File: go.mod [hunk 1]
```diff
@@ -4,8 +4,15 @@ go 1.16
 
 require (
 	github.com/antlr/antlr4/runtime/Go/antlr v0.0.0-20210521073959-f0d4d129b7f1
+	github.com/golang/mock v1.6.0
+	github.com/google/go-cmp v0.5.4 // indirect
+	github.com/google/uuid v1.2.0
+	github.com/nspcc-dev/neo-go v0.95.3
```
[the remark this answers is not in the document]
Comment (KirillovDenis) [reply go.mod]: Fixed

**Pull request merged**